crates/uv/src/commands/run.rs

```diff
@@ -16,16 +16,28 @@ pub(crate) async fn run_to_completion(mut handle: Child) -> anyhow::Result<ExitS
         use tokio::signal::unix::{signal, SignalKind};
 
         let mut term_signal = signal(SignalKind::terminate())?;
+        let mut int_signal = signal(SignalKind::interrupt())?;
         loop {
             select! {
                 result = handle.wait() => {
                     break result;
                 },
 
+                // TODO(zanieb: Refactor `interrupt_process` and `terminate_process` to use
+                // shared logic.
+                // `SIGINT`
+                _ = int_signal.recv() => {
+                    int_signal_count += 1;
+                    if int_signal_count > 1 {
+                        let _ = interrupt_process(&mut handle);
+                    }
+                },
```

Comment:
I'd like to understand why Ctrl-C isn't working in https://github.com/astral-sh/uv/issues/10952 but works in other cases, but... it seems generally correct to forward a SIGINT if we've received it multiple times (as an escape hatch).